terraform/init/variables.tf

```diff
@@ -0,0 +1,12 @@
+variable "prefix" {
+  default       = "test"
+  description   = "Prefix of the resource group name that's combined with a random ID so name is unique in your Azure subscription."
+}
```

Comment:
Can the `prefix` be dynamic? I am imagining that we name all the resources based on the branch's name. If so, do we:
1. Still need the random id? That is, if there's a branch 1 year ago named `foo` which has been merged into `master`, and now I create a branch `foo`, will there be an issue?
2. Do resources need to have a globally unique name (i.e. unique across all resources created by everyone using Azure). If not, can we simply name the Resource Group with the branch's name, and all the resources inside the Resource Group do not use prefixes? (i.e. imagine a folder containing files - the files across the different folders can have the same name).